var/spack/repos/builtin/packages/ucc/package.py

```diff
@@ -0,0 +1,49 @@
+# Copyright 2013-2023 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+from spack.package import *
+
+
+class Ucc(AutotoolsPackage, CudaPackage):
+    """UCC is a collective communication operations API and library that is
+    flexible, complete, and feature-rich for current and emerging programming
+    models and runtimes."""
+
+    homepage = "https://openucx.github.io/ucc/"
+    url = "https://github.com/openucx/ucc/archive/refs/tags/v1.1.0.tar.gz"
+
+    # maintainers("github_user1", "github_user2")
+
+    version("1.2.0", sha256="c1552797600835c0cf401b82dc89c4d27d5717f4fb805d41daca8e19f65e509d")
+    version("1.1.0", sha256="74c8ba75037b5bd88cb703e8c8ae55639af3fecfd4428912a433c010c97b4df7")
+    version("1.0.0", sha256="d3b4aa7004bf339d35952a1699a6e408064ba578bdc93861f5f07527ad0a5e8c")
+
+    variant("cuda", default=False, description="Enable CUDA TL")
+    variant("nccl", default=False, description="Enable NCCL TL")
+    variant("rccl", default=False, description="Enable RCCL TL", when="@1.1:")
+    variant("ucp", default=True, description="Enable UCP TL")
+
+    conflicts("+cuda", when="@:1.0", msg="UCC CUDA TL added in version 1.1")
+    conflicts("cuda@12:", when="@1.1", msg="UCC 1.1 supports CUDA <12")
+    conflicts("~cuda", when="+nccl", msg="UCC NCCL TL requires CUDA")
```

Comment:
```suggestion
    variant("cuda", default=False, when="@1.1:", description="Enable CUDA TL")
    variant("nccl", default=False, description="Enable NCCL TL")
    variant("rccl", default=False, description="Enable RCCL TL", when="@1.1:")
    variant("ucp", default=True, description="Enable UCP TL")

    conflicts("cuda@12:", when="@1.1", msg="UCC 1.1 supports CUDA <12")
    conflicts("~cuda", when="+nccl", msg="UCC NCCL TL requires CUDA")
```

Reply:
The [CudaPackage documentation](https://spack.readthedocs.io/en/latest/build_systems/cudapackage.html#conflicts) suggests conflicts instead of conditional variants, I thought there was some gotcha behind it.